src/nanoarrow/ipc/decoder.c

```diff
@@ -1485,14 +1580,27 @@ static ArrowErrorCode ArrowIpcMakeBufferFromShared(struct ArrowIpcBufferFactory*
                                                    struct ArrowBufferView* dst_view,
                                                    struct ArrowBuffer* dst,
                                                    struct ArrowError* error) {
-  NANOARROW_UNUSED(error);
-
   struct ArrowIpcSharedBuffer* shared =
       (struct ArrowIpcSharedBuffer*)factory->private_data;
-  ArrowBufferReset(dst);
-  ArrowIpcSharedBufferClone(shared, dst);
-  dst->data += src->body_offset_bytes;
-  dst->size_bytes = src->buffer_length_bytes;
+
+  int needs_decompression = 0;
```

Comment:
Ultra nit but is there any reason why we shy away from using stdbool?

Reply:
Sounds good. Not important enough to change I think. 

Can also wait 20 years for C23 to be fully supported by MSVC